src/zabapgit_repo_impl.prog.abap

```diff
@@ -533,7 +535,20 @@ CLASS lcl_repo IMPLEMENTATION.
 
     lt_cache = mt_local.
     lt_tadir = lcl_tadir=>read( get_package( ) ).
-    LOOP AT lt_tadir ASSIGNING <ls_tadir>.
+
+    IF it_filter[] IS INITIAL.
```

Comment:
I just implemented this. Thanks for the comments. Before being pulled please somebody answer this from #553:

> What I've noted so far and need to clarify before creating a PR is that in the actual situation, a ".abapgit.xml" is not being generated (because it is deleted when filter is applied). With the proposed solution the ".abapgit.xml" is being generated and not deleted. So, what I'm not sure of, is that if that file is necessary to add to the zip file.